spec/spec_helper_acceptance.rb

```diff
@@ -26,6 +26,30 @@ def default_installed_bin_dir
   end
 end
 
+# This method allows a block to be passed in and if an exception is raised
+# that matches the 'error_matcher' matcher, the block will wait a set number
+# of seconds before retrying.
+# Params:
+# - max_retry_count - Max number of retries
+# - retry_wait_interval_secs - Number of seconds to wait before retry
+# - error_matcher - Matcher which the exception raised must match to allow retry
+# Example Usage:
+# retry_on_error_matching(3, 5, /OpenGPG Error/) do
+#   apply_manifest(pp, :catch_failures => true)
+# end
+def retry_on_error_matching(max_retry_count = 3, retry_wait_interval_secs = 5, error_matcher = nil)
+  try = 0
+  begin
+    try += 1
+    yield
+  rescue StandardError => e
+    raise unless try < max_retry_count && (error_matcher.nil? || e.message =~ error_matcher)
```

Comment:
we should probably add an error matcher so we don't get unexpected behaviour with `error_matcher.nil?` or mask errors we aren't expecting

Reply:
Tried that and the `expected File to exist`

Reply:
Also, is that really an error?? At a glance it doesn't look like it has anything to do with the test and is just the last thing in the output before it tried to find the file and couldn't??

Reply:
To me, reading through the tests, that bit of output just reads as a part of the standard info dump that the PDK makes.

Reply:
yeah my thoughts too.. Alrighty if theres no real error matcher you can supply its all good